### PR TITLE
Intermediate Core downstream / prepare tagging upstream

### PR DESCRIPTION
**Pull request by CoMPaTech**
Points to notice:
 - [ ] We must check and thoroughly test the zeroconf change in manifest (not sure if this works in custom_components)
 - [ ] Mostly mentions of consts, but we might want to validate before upstreaming
 - [ ] Downstream most of the add_entities constructors
 - [ ] Language still to compare
 - [ ] Tests still to compare
 - [ ] `climate.py` has issues when rewriting as separate pw-beta and core functionality (rough progress in gist; https://gist.github.com/CoMPaTech/664d3d15a4572be5769fb510df154f2b)
 - [ ] We need to check alignment on the `Set up [Plugwise/Smile] ...` descriptors :)
 - [ ] Added parallels on all platforms (as induced from https://developers.home-assistant.io/docs/integration_fetching_data?_highlight=paral#request-parallelism)

## Summary by CodeRabbit

- **New Features**
  - Enhanced automatic discovery of Plugwise devices through updated integration metadata.

- **Improvements**
  - Updated integration name to "Plugwise Beta" for simplicity and clarity.
  - Added support for button controls in the integration, expanding its functionality.
  - Updated Home Assistant version requirement to ensure compatibility with the latest features and fixes.
  - Adjusted update handling in the Plugwise Climate component to improve stability.

- **Documentation**
  - Updated the CHANGELOG to reflect recent changes and improvements, providing better insights into updates.

### CODE REVIEW

**Review by bouwew — COMMENTED** [review]
Please remove the `async_setup_entry()`-changes, I kept the code like it was because I wanted to keep in the debug-logging for troubleshooting.

File: custom_components/plugwise/__init__.py [hunk 1]
```diff
@@ -71,9 +76,9 @@ async def delete_notification(
             )
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
-    entry.async_on_unload(entry.add_update_listener(update_listener))
 
-    for component in PLATFORMS:  # pw-beta
+    entry.async_on_unload(entry.add_update_listener(update_listener))  # pw-beta delete_notification
```
Comment (bouwew) [inline custom_components/plugwise/__init__.py:80]: The `update_listener` is there for the `OptionsConfigFlow`.

File: custom_components/plugwise/binary_sensor.py [hunk 2]
```diff
@@ -93,36 +96,33 @@ async def async_setup_entry(
...
+            if (
+                binary_sensors := coordinator.data.devices[device_id].get(
+                    BINARY_SENSORS  # upstream to const
                 )
-
-        async_add_entities(entities)
+            )
+            for description in PLUGWISE_BINARY_SENSORS  # upstream from const
+            if description.key in binary_sensors
+        )
```
Comment (bouwew) [inline custom_components/plugwise/binary_sensor.py:118]: I kept the code like this on purpose because I wanted to keep the debug-logging.

**Comment by CoMPaTech** [comment]
> Please remove the `async_setup_entry()`-changes, I kept the code like it was because I wanted to keep in the debug-logging for troubleshooting.

Reverted but dropped ATTR_NAME just for clarity, tried wrapping/overloading the Core functionality but that would take considerable more mypy time. If it is better for diagnostics we might challenge this to core as well by adding debug check to that helpers class? (Outside of scope for this PR though)

File: custom_components/plugwise/__init__.py [hunk 3]
```diff
@@ -1,30 +1,35 @@
...
+    SERVICE_DELETE,  # pw-beta delete_notifications
 )
 from .coordinator import PlugwiseDataUpdateCoordinator
 
 type PlugwiseConfigEntry = ConfigEntry[PlugwiseDataUpdateCoordinator]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: PlugwiseConfigEntry) -> bool:
-    """Set up the Plugwise Device from a config entry."""
+    """Set up Plugwise components from a config entry."""
```
Comment (bouwew) [inline custom_components/plugwise/__init__.py:32]: Should we write here: `"""Set up the Plugwise Integration from...`?

File: custom_components/plugwise/__init__.py [hunk 4]
```diff
@@ -88,7 +93,7 @@ async def update_listener(
     await hass.config_entries.async_reload(entry.entry_id)
 
 async def async_unload_entry(hass: HomeAssistant, entry: PlugwiseConfigEntry) -> bool:
-    """Unload a config entry."""
+    """Unload the Plugwise components."""
```
Comment (bouwew) [inline custom_components/plugwise/__init__.py:96]: Also here: `"""Unload the Plugwise Integration.`? Or `""" Unload the Plugwise platforms.`? Or `"""Unload the Plugwise config entry.`? The last two seem to make the most sense, to me :)

File: custom_components/plugwise/__init__.py [hunk 5]
```diff
@@ -125,15 +129,21 @@ def migrate_sensor_entities(
...
         if entity_id := ent_reg.async_get_entity_id(
             Platform.SENSOR, DOMAIN, old_unique_id
         ):
             new_unique_id = f"{device_id}-outdoor_air_temperature"
+            LOGGER.debug(
+                "Migrating entity %s from old unique ID '%s' to new unique ID '%s'",
+                entity_id,
+                old_unique_id,
+                new_unique_id,
+            )
```
Comment (bouwew) [inline custom_components/plugwise/__init__.py:148]: Let's get rid of this debug-message.

File: custom_components/plugwise/binary_sensor.py [hunk 6]
```diff
@@ -93,15 +97,29 @@ async def async_setup_entry(
...
+        #     if (
+        #         binary_sensors := coordinator.data.devices[device_id].get(
+        #             BINARY_SENSORS
+        #         )
+        #     )
+        #     for description in PLUGWISE_BINARY_SENSORS
+        #     if description.key in binary_sensors
+        # )
+
+        # pw-beta alternative for debugging
```
Comment (bouwew) [inline custom_components/plugwise/binary_sensor.py:122]: I don't mind this :)

File: custom_components/plugwise/climate.py [hunk 7]
```diff
@@ -234,13 +231,9 @@ def hvac_action(self) -> HVACAction:  # pw-beta add to Core
...
 
         # Adam provides the hvac_action for each thermostat
-        if (control_state := self.device.get(CONTROL_STATE)) == HVACAction.COOLING:
-            return HVACAction.COOLING
-        if control_state == HVACAction.HEATING:
-            return HVACAction.HEATING
-        if control_state == HVACAction.PREHEATING:
-            return HVACAction.PREHEATING
-        if control_state == STATE_OFF:
+        if (control_state := self.device.get(CONTROL_STATE)) in (HVACAction.COOLING, HVACAction.HEATING, HVACAction.PREHEATING):
```
Comment (bouwew) [inline custom_components/plugwise/climate.py:234]: Nice!

File: custom_components/plugwise/climate.py [hunk 8]
```diff
@@ -234,13 +231,9 @@ def hvac_action(self) -> HVACAction:  # pw-beta add to Core
...
+            return control_state
+        if control_state == HVACMode.OFF:
```
Comment (bouwew) [inline custom_components/plugwise/climate.py:236]: Does this work with mypy?

File: custom_components/plugwise/climate.py [hunk 9]
```diff
@@ -292,7 +285,7 @@ async def async_set_hvac_mode(self, hvac_mode: str) -> None:
         if hvac_mode != HVACMode.OFF:
             await self.coordinator.api.set_schedule_state(
                 self.device[LOCATION],
-                STATE_ON if hvac_mode == HVACMode.AUTO else STATE_OFF,
+                STATE_ON if hvac_mode == HVACMode.AUTO else HVACMode.OFF,
```
Comment (bouwew) [inline custom_components/plugwise/climate.py:288]: Sorry, I don't like this. Better to keep the schedule state separated from HVACModes.

File: custom_components/plugwise/manifest.json [hunk 10]
```diff
@@ -1,13 +1,13 @@
...
   "codeowners": ["@CoMPaTech", "@bouwew"],
   "config_flow": true,
   "documentation": "https://github.com/plugwise/plugwise-beta",
   "integration_type": "hub",
   "iot_class": "local_polling",
   "loggers": ["plugwise"],
   "requirements": ["plugwise==0.38.3"],
-  "version": "0.51.6"
+  "version": "0.51.6",
+  "zeroconf": ["_plugwise._tcp.local."]
```
Comment (bouwew) [inline custom_components/plugwise/manifest.json:12]: My first response would be: don't add this line! Pw-beta counts on the Core Plugwise integration doing this.
But you're right, if the Core Plugwise integration has never been loaded before, this is probably needed.

File: custom_components/plugwise/climate.py (hunk 8 above)
Comment (CoMPaTech) [reply custom_components/plugwise/climate.py:236]: The strange thing is some `.get` constructs do work in Core but not in beta ...

File: custom_components/plugwise/__init__.py (hunk 5 above)
Comment (CoMPaTech) [reply custom_components/plugwise/__init__.py:148]: It's still in core, adjusting and flagging accordingly

File: custom_components/plugwise/__init__.py (hunk 3 above)
Comment (CoMPaTech) [reply custom_components/plugwise/__init__.py:32]: It's a complete mixed bag ... I've rewritten most of them to `Set up Plugwise x from config entry` now though with your next comment we might need more of those. Core is a mix of Plugwise and Smile and ConfigEntry vs config entry :)
Maybje just `Setup Plugwise from a config Entry`?

File: custom_components/plugwise/__init__.py (hunk 4 above)
Comment (CoMPaTech) [reply custom_components/plugwise/__init__.py:96]: Upload plugwise (also followed Core on this one)

File: custom_components/plugwise/climate.py (hunk 8 above)
Comment (bouwew) [reply custom_components/plugwise/climate.py:242]: I meant the `control_state` - `str` vs `HVACMode.Off` - `StrEnum`.

File: custom_components/plugwise/__init__.py (hunk 3 above)
Comment (bouwew) [reply custom_components/plugwise/__init__.py]: Maybe `Set up the Plugwise Integration from a config entry`?
Good to add `config entry` :+1:

File: custom_components/plugwise/__init__.py (hunk 4 above)
Comment (bouwew) [reply custom_components/plugwise/__init__.py]: `Unload the Plugwise config entry`? Fits nicely with the `async_setup_entry()` docstring.

**Comment by CoMPaTech** [comment]
In 'our' HA core there is a `pwsmall` branch fwiw (not yet fully up to par)

**Review by bouwew — APPROVED** [review]
Ok, let's merge these updates!

**Comment by bouwew** [comment]
I added the zeroconf-line in my system, all looks normal after a HA restart.

**Comment by CoMPaTech** [comment]
Confirmed for merge?

**Pull request merged**